code/game/machinery/airlock_control.dm

```diff
@@ -87,13 +87,18 @@
 /obj/machinery/door/airlock/Destroy()
 	if(frequency)
 		SSradio.remove_object(src,frequency)
-	UnregisterSignal(src, COMSIG_GREY_TIDE)
+	UnregisterSignal(SSdcs, COMSIG_GLOB_GREY_TIDE)
 	return ..()
 
 /obj/machinery/door/airlock/proc/grey_tide()
-	if(critical_machine) //Skip doors in critical positions, such as the SM chamber.
-		return
-	prison_open()
+	SIGNAL_HANDLER
+
+	if(is_station_level(z))
+		for(var/area_type in GLOB.grey_tide_areas)
+			if(istype(get_area(src), area_type))
+				if(critical_machine) //Skip doors in critical positions, such as the SM chamber.
+					return
+				INVOKE_ASYNC(src, PROC_REF(prison_open), 2) //Sleep gets called further down in open(), so we have to invoke async
```

Comment:
Named parameter for 2. Because this is INVOKE_ASYNC, it isn't supported, but you can do `/* name = */ 2`